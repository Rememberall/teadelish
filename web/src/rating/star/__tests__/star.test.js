// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Star, { getFill } from '../index';

describe('fill color', () => {
  it('should be none when there is no selected rating', () => {
    expect(getFill(3, undefined)).toBe('none');
  });

  it('should be gold when the selected rating is > the rating', () => {
    expect(getFill(3, 4)).toBe('gold');
  });

  it('should be gold when the selected rating is = the rating', () => {
    expect(getFill(4, 4)).toBe('gold');
  });

  it('should be lightgrey when the selected rating is < the rating', () => {
    expect(getFill(4, 3)).toBe('lightgrey');
  });
});

describe('snapshots', () => {
  it('should not change', () => {
    [undefined, null, 1, 2, 3, 4, 5].forEach(selectedRating => {
      const tree = renderer
        .create(<Star onSelect={() => {}} selectedRating={selectedRating} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
