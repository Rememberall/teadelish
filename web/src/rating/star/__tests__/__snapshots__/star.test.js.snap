// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should not change 1`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;

exports[`snapshots should not change 2`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="none"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;

exports[`snapshots should not change 3`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={true}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;

exports[`snapshots should not change 4`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={true}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;

exports[`snapshots should not change 5`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={true}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;

exports[`snapshots should not change 6`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={true}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="lightgrey"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;

exports[`snapshots should not change 7`] = `
<div
  className="star-rating"
>
  <button
    className="star-rating-clear"
    onClick={[Function]}
  >
    Clear
  </button>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={false}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
  <label>
    <svg
      height="45"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Five Pointed Star
      </title>
      <path
        d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
        fill="gold"
        stroke="black"
      />
      <div
        xmlns="http://www.w3.org/1999/xhtml"
      />
    </svg>
    <input
      checked={true}
      name="selected-rating"
      onChange={[Function]}
      type="radio"
    />
  </label>
</div>
`;
